### PR TITLE
BlockPos mappings

### PR DESCRIPTION
**Pull request by liach**
Signed-off-by: liach <liach@users.noreply.github.com>

### CODE REVIEW

**Review by YanisBft — APPROVED** [review]

**Review by jaskarth — APPROVED** [review]

File: mappings/net/minecraft/util/math/BlockPos.mapping [hunk 1]
```diff
@@ -69,28 +69,89 @@ CLASS net/minecraft/class_2338 net/minecraft/util/math/BlockPos
...
 		ARG 0 dynamic
 	METHOD method_20437 stream (Lnet/minecraft/class_2338;Lnet/minecraft/class_2338;)Ljava/util/stream/Stream;
-		ARG 0 pos1
-		ARG 1 pos2
+		ARG 0 start
+		ARG 1 end
 	METHOD method_23627 stream (Lnet/minecraft/class_3341;)Ljava/util/stream/Stream;
 		ARG 0 box
 	METHOD method_25503 mutableCopy ()Lnet/minecraft/class_2338$class_2339;
+	METHOD method_25996 manhattanIterate (Lnet/minecraft/class_2338;III)Ljava/lang/Iterable;
```
Comment (Runemoro) [inline mappings/net/minecraft/util/math/BlockPos.mapping]: I think a better name is `iterateOutwardsManhattan`, which makes it very clear what it does.

File: mappings/net/minecraft/util/math/BlockPos.mapping [hunk 2]
```diff
@@ -69,28 +69,89 @@ CLASS net/minecraft/class_2338 net/minecraft/util/math/BlockPos
...
+		ARG 2 yRadius
+			COMMENT the maximum y difference for iteration
+		ARG 3 zRadius
+			COMMENT the maximum z difference for iteration
+	METHOD method_25997 findClosest (Lnet/minecraft/class_2338;IILjava/util/function/Predicate;)Ljava/util/Optional;
+		ARG 0 pos
+		ARG 1 horizontalRange
+		ARG 2 verticalRange
+		ARG 3 condition
+	METHOD method_25998 manhattanStream (Lnet/minecraft/class_2338;III)Ljava/util/stream/Stream;
```
Comment (Runemoro) [inline mappings/net/minecraft/util/math/BlockPos.mapping]: `streamOutwardsManhattan`

File: mappings/net/minecraft/util/math/BlockPos.mapping [hunk 3]
```diff
@@ -69,28 +69,89 @@ CLASS net/minecraft/class_2338 net/minecraft/util/math/BlockPos
...
+		FIELD field_23108 startX I
+		FIELD field_23109 startY I
+		FIELD field_23110 startZ I
+		FIELD field_23111 index I
 	CLASS class_2339 Mutable
-		METHOD method_10098 setOffset (Lnet/minecraft/class_2350;)Lnet/minecraft/class_2338$class_2339;
+		METHOD method_10098 move (Lnet/minecraft/class_2350;)Lnet/minecraft/class_2338$class_2339;
 			ARG 1 direction
-		METHOD method_10100 setOffset (III)Lnet/minecraft/class_2338$class_2339;
+		METHOD method_10100 move (III)Lnet/minecraft/class_2338$class_2339;
```
Comment (Runemoro) [inline mappings/net/minecraft/util/math/BlockPos.mapping]: `move` can be understood as moving to the given possition. `offset` or `shift` would be clearer.

Comment (liach) [reply mappings/net/minecraft/util/math/BlockPos.mapping]: It is moving by the given parameters. `moveTo` would be understand as `set` instead.

File: mappings/net/minecraft/util/math/BlockPos.mapping (hunk 2 above)
Comment (liach) [reply mappings/net/minecraft/util/math/BlockPos.mapping]:
```suggestion
	METHOD method_25998 streamOutwards (Lnet/minecraft/class_2338;III)Ljava/util/stream/Stream;
```

File: mappings/net/minecraft/util/math/BlockPos.mapping (hunk 1 above)
Comment (liach) [reply mappings/net/minecraft/util/math/BlockPos.mapping]:
```suggestion
	METHOD method_25996 iterateOutwards (Lnet/minecraft/class_2338;III)Ljava/lang/Iterable;
```

File: mappings/net/minecraft/util/math/BlockPos.mapping [hunk 4]
```diff
@@ -69,28 +69,89 @@ CLASS net/minecraft/class_2338 net/minecraft/util/math/BlockPos
...
 			ARG 1 x
 			ARG 2 y
 			ARG 3 z
```
Comment (liach) [inline mappings/net/minecraft/util/math/BlockPos.mapping]:
```suggestion
			ARG 1 dx
			ARG 2 dy
			ARG 3 dz
```

**Review by Juuxel — APPROVED** [review]

**Review by Runemoro — APPROVED** [review]

**Pull request merged**